app/views/order_items/index.html.erb

```diff
@@ -0,0 +1,32 @@
+<h1>Your Cart</h1>
+
+  <ul>
+    <% total = 0 %>
+    <% @order_items.each do |item| %>
+      <% product = Product.find(item.product_id) %>
+      <li>
+        <%= product.name %>
+        <ul>
+          <li>Price: <%= product.price %></li>
+          <li>Quantity: <%= item.quantity %>
+            <% if product.stock > 1 %>
+              <%= form_for(item, url: update_cart_path(item.id), method: :patch) do |f| %>
+                <%= f.label "Change Quantity" %>
+                <%= f.select(:quantity, options_for_select((1..product.stock)), include_blank: true) %>
+                <%= f.submit "Update" %>
+              <% end %>
+            <% else %>
+              (Only 1 left in stock!)
+            <% end %>
+          </li>
+          <li><%= link_to "Remove This Item", {:controller => :order_items, :action => 'destroy', :id => item.id}, method: :delete %></li>
+        </ul>
+      </li>
+      <% total += (product.price * item.quantity) %>
```

Comment:
The logic of calculating the total should not be in the view. The view should generally be displaying information, not calculating it. Instead this could be a method, `calculate_total`, in the order_item model.

Also, when this logic is in the view you can't really test it. Making it a method in the model allows you to write tests to make sure it is calculating correct totals for your orders.